obelix/idefix/services/google/gcalendar.go

```diff
@@ -0,0 +1,116 @@
+package google
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/Shopify/sarama"
+	redisv8 "github.com/go-redis/redis/v8"
+	"google.golang.org/api/calendar/v3"
+	"idefix/operator"
+	"idefix/producer"
+	"idefix/redis"
+	"log"
+	"time"
+)
+
+const (
+	ACTIVE = iota
+	NO_ACTIVE
+)
+
+type GCalendar struct {
+	date string
+}
+
+func (gc *GCalendar) Parse(srv *calendar.Service) error {
+	t := time.Now().Format(time.RFC3339)
+	events, err := srv.Events.List("primary").ShowDeleted(false).SingleEvents(true).TimeMin(t).MaxResults(1).OrderBy("startTime").Do()
+	if err != nil {
+		return fmt.Errorf("failed to retrieve next event gcalendar: %v", err)
+	}
```

Comment:
Consider moving these lines in a separate method.

Reply:
6572d568177a397940d5454fb1d43e8c72a00a97